infra-as-code/bicep/modules/vwanConnectivity/vwanConnectivity.bicep

```diff
@@ -0,0 +1,199 @@
+
+@description('Prefix Used for ExpressRoute Gateway. Default: {parCompanyPrefix}-ergw-{resourceGroup().location}')
+param parERGwName string = '${parCompanyPrefix}-ergw-${resourceGroup().location}'
+
+@description('Azure Firewall Name. Default: {parCompanyPrefix}-fw-{resourceGroup().location}')
+param parAzureFirewallName string = '${parCompanyPrefix}-fw-${resourceGroup().location}'
+
+@description('Region in which the resource group was created. Default: {resourceGroup().location}')
+param parLocation string = resourceGroup().location
+
+@description('The scale unit for this VPN Gateway: Default: 1')
+param parVPNGwScaleUnit int = 1
+
+@description('The scale unit for this ExpressRoute Gateway: Default: 1')
+param parERGwScaleUnit int = 1
+
+@description('Set Parameter to true to Opt-out of deployment telemetry')
+param parTelemetryOptOut bool = false
+
+// Customer Usage Attribution Id
+var varCuaid = '7f94f23b-7a59-4a5c-9a8d-2a253a566f61'
+
+// Virtual WAN resource
+resource resVWAN 'Microsoft.Network/virtualWans@2021-05-01' = {
+  name: parVWanName
+  location: parLocation
+  tags: parTags
+  properties: {
+    allowBranchToBranchTraffic: true
+    allowVnetToVnetTraffic: true
+    disableVpnEncryption: false
+    type: 'Standard'
+  }
+}
+
+resource resVHub 'Microsoft.Network/virtualHubs@2021-05-01' = if(parVirtualHubEnabled && !empty(parVhubAddressPrefix)) {
+  name: parVHubName
+  location: parLocation
+  tags: parTags
+  properties: {
+    addressPrefix: parVhubAddressPrefix
+    sku: 'Standard'
+    virtualWan: {
+      id: resVWAN.id
+    }
+  }
+}
+
+resource resVPNGateway 'Microsoft.Network/vpnGateways@2021-05-01' = if(parVirtualHubEnabled && parVPNGatewayEnabled) {
+  name: parVPNGwName
+  location: parLocation
+  tags: parTags
+  properties: {
+    bgpSettings: {
+      asn: 65515
+      bgpPeeringAddress: ''
+      peerWeight: 5
+    }
+    virtualHub:{
+      id: resVHub.id
+    }
+    vpnGatewayScaleUnit: parVPNGwScaleUnit
+  }
+}
+
+resource resERGateway 'Microsoft.Network/expressRouteGateways@2021-05-01' = if (parVirtualHubEnabled && parERGatewayEnabled) {
+  name: parERGwName
+  location: parLocation
+  tags: parTags
+  properties: {
+    virtualHub:{
+      id: resVHub.id
+    }
+    autoScaleConfiguration: {
+      bounds: {
+        min: parERGwScaleUnit
+      }
+    }
+  }
+}
+
+// AzureFirewallSubnet is required to deploy Azure Firewall . This subnet must exist in the parsubnets array if you deploy.
+// There is a minimum subnet requirement of /26 prefix.  
+resource resAzureFirewall 'Microsoft.Network/azureFirewalls@2021-02-01' = if (parVirtualHubEnabled && parAzureFirewallEnabled) {
+  name: parAzureFirewallName
+  location: parLocation
+  tags: parTags
+  dependsOn: [
+    // putting an explicit dependency because there is an implicit dependency below which produces a deployment error if parAzureFirewallEnabled == false
+    modAzureFirewallPublicIP
```

Comment:
see below fix for this on line 154 👍

This explicit dependsOn can be removed 👍

Reply:
ack. removed explicit dependsOn